openmdao/visualization/meta_model_viewer/meta_model_visualization.py

```diff
@@ -0,0 +1,551 @@
+        
+        outputs = {i : [] for i in self.output_list}
+        print("Making Predictions")
+
+        # Parse dict into shape [n**2, number of inputs] list
+        inputs = np.empty([self.resolution**2, self.nx])
+        for idx, values in enumerate(data.values()):
+            inputs[:, idx] = values.flatten()
+
+        # Pair data points with their respective prob name. Loop to make predictions
+        for idx, tup in enumerate(inputs):
+            for name, val in zip(data.keys(), tup):
+                self.prob[self.surrogate_comp.name + '.' + name] = val
+            self.prob.run_model()
+            for i in self.output_list:
+                outputs[i].append(float(self.prob[self.surrogate_comp.name + '.' + i]))
+
+        return self.stack_outputs(outputs)
+
+    def contour_data(self):
+
+        """
+        Creates a contour plot.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        Bokeh Image Plot
+        """
+
+        n = self.resolution
+        xe = np.zeros((n, n, self.nx))
+        ye = np.zeros((n, n, self.ny))
+
+        # Query the slider dictionary, append the name and current value to the ordered dictionary
+        self.slider_value_and_name = OrderedDict()
+        for title, slider_params in self.slider_dict.items():
+            self.slider_value_and_name[title] = slider_params.value
+        
+        # Cast the current values of the slider_value_and_name dictionary values to a list
+        self.input_point_list = list(self.slider_value_and_name.values())
+        for ix in range(self.nx):
+            xe[:, :, ix] = self.input_point_list[ix]
+
+        # Search the input_data_dict to match the names with the x/y dropdown menus. Then set x/y
+        # linspaces to the values for the meshgrid which follows
+        for title, values in self.input_data_dict.items():
+            if title == self.x_input.value:
+                xlins = values
+                dw = max(values)
+            if title == self.y_input.value:
+                ylins = values
+                dh = max(values)
+
+        # Create a mesh grid and then append that data to 'xe' in the respective columns
+        X, Y = np.meshgrid(xlins, ylins)
+        xe[:, :, self.x_index] = X
+        xe[:, :, self.y_index] = Y
+
+        # This block places the x and y inputs first and then appends any 
+        # other values to the list the first two points
+        pred_dict = {}
+        self.input_list = [self.x_input.value, self.y_input.value]
+        for title in self.slider_value_and_name.keys():
+            if title == self.x_input.value or title == self.y_input.value:
+                pass
+            else:
+                self.input_list.append(title)
+        
+        # Append the key (input_list) and the values copied from xe to pred_dict 
+        # where it is then ordered in pred_dict_ordered.
+        for idx, title in enumerate(self.slider_value_and_name.keys()):
```

Comment:
I think you can just do this without the for loop by passing `self.slider_value_and_name.items()` directly to the update method on the dict

Reply:
I can't do that because I rely on the idx from enumerate to append the data from `x_data` to the dict values.